robotoff/app/api.py

```diff
@@ -1055,6 +1055,51 @@ def on_get(self, req: falcon.Request, resp: falcon.Response, username: str):
         resp.media = {"count": {"annotations": annotation_count}}
 
 
+class DisplayInsightPredictionForProducts:
+    def on_get(self, req: falcon.Request, resp: falcon.Response):
+        response: JSONType = {}
+        count: int = req.get_param_as_int("count", min_value=1, default=25)
+        keep_types: Optional[List[str]] = req.get_param_as_list(
+            "insight_types", required=False
+        )
+        barcode: Optional[str] = req.get_param("barcode")
+        value_tag: str = req.get_param("value_tag")
+        brands = req.get_param_as_list("brands") or None
+        server_domain: Optional[str] = req.get_param("server_domain")
+
+        if keep_types:
+            # Limit the number of types to prevent slow SQL queries
+            keep_types = keep_types[:10]
+
+        if brands is not None:
+            # Limit the number of brands to prevent slow SQL queries
+            brands = brands[:10]
```

Comment:
I think we can factor this code between InsightCollection and DisplayInsightPredictionForProducts.

The idea, is that the function can return a dict (say `query_params`), and then simply use:
`get_predictions = functools.partial(get_predictions, **query_params)`